### DISCUSSION TITLE
MAUI : Offset function for a MRect

### DISCUSSION THREAD

**Discussion by Elsassma** [Mapsui/Mapsui#2682] [Q&A] [answered]
Hello, 

I am migrating an app from Xamarin to MAUI and I use the MAUI Mapsui library.
Everything works nice except one thing where I was not able to find a replacment : the Offset function...

In Xamarin I have this code : 
```csharp
Mapsui.Geometries.BoundingBox boundingBox = new Mapsui.Geometries.BoundingBox(0, 0, maxX, maxY);
boundingBox.Offset(envelope.Centroid - boundingBox.Centroid);
```

However, in MAUI, the MRect (which replaces the BoundingBox) doesn't have an Offset function. 
What is the way to do to replace this function?

**Comment by pauldendulk** [comment, marked as answer]
This is the old implementation:

https://github.com/Mapsui/Mapsui/blob/1fb47842c38b229a43fdf359aa3f34705838423c/Mapsui.Geometries/BoundingBox.cs#L196

You could create an Offset extension method to update the Min.X/Min.Y/Max.X/Max.Y values.

Or perhaps better let is return an updated copy of the MRect because in a future version the MPoint and MRect might become immutable.

**Reply by Elsassma** [reply]
Thanks for the answer, it works as I wanted.
